egs/wsj/s5/utils/data/perturb_data_dir_volume.sh

```diff
@@ -68,6 +82,19 @@ if $volume_perturb_done; then
   exit 0
 fi
 
+if [ $perturb_mode == absvol ]; then
+  reco2vol=$data/reco2vol
+  wav-to-volume scp:$data/wav.scp ark,t:- | awk -v awk_seed=${local_seed}  \
+    -v awk_low=${abs_low} -v awk_high=${abs_high}                          \
+    -v awk_out=${reco2vol}                                                 \
+    'BEGIN{ srand(awk_seed); }                                             \
+    {                                                                      \
+      dst_vol=int(rand() * (awk_high - awk_low) + awk_low);                \
+      scale=(dst_vol * 1.0) / ($2 * 1.0);                                  \
+      print $1,scale > awk_out;                                            \
+    }'
```

Comment:
Wouldn't this be quite slow, since it would have to read all the wav files without parallelization?
Also, I think you can redirect the awk command to that file instead of doing the '> awk_out' inside the awk script, that's simpler.
But I'll ask separately for more details about why you are adding this.